### PR TITLE
Update storage blob SDK for java from track1 to track2

### PR DESCRIPTION
**Pull request by v-hongli1**
1. Update storage blob SDK for java from track1 to track2
2. Changed files:
    1) storage-blobs-java-v10-tutorial-master/pom.xml
    2) storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java

@scottaddie, @joshfree, @tg-msft, @kyle-patterson and @alzimmermsft for notification

### CODE REVIEW

File: storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java [hunk 1]
```diff
@@ -49,109 +38,72 @@ static File createTempFile() throws IOException {
...
-                System.out.println("Completed upload request.");
-                System.out.println(response.response().statusCode());
-            });
-
-            
+    static void uploadFile(BlockBlobAsyncClient blob, File sourceFile) throws IOException {
+        ByteBuffer buffer = ByteBuffer.allocate(8 * 1024 * 1024);
+        AsynchronousFileChannel fileChannel = AsynchronousFileChannel.open(sourceFile.toPath());
+        fileChannel.read(buffer, 0);
+        blob.uploadWithResponse(new BlockBlobSimpleUploadOptions(
```
Comment (alzimmermsft) [inline storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java]: There are upload from file APIs that could be used instead: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java#L395

File: storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java [hunk 2]
```diff
@@ -49,109 +38,72 @@ static File createTempFile() throws IOException {
...
-        // BlobRange is defined from 0 to 4MB
-        blobURL.download(new BlobRange().withOffset(0).withCount(4*1024*1024L), null, false, null)
-                .flatMapCompletable(response -> {
-                    if(response.rawResponse().request().url().getHost().contains("-secondary"))
-                    {
+    static void getBlob(BlockBlobAsyncClient blockBlobAsyncClient, File sourceFile) {
+        blockBlobAsyncClient.downloadStreamWithResponse(
+                new BlobRange(0, 4 * 1024 * 1024L),
+                null, null, false)
+                .doOnSubscribe(onSubscribe -> System.out.println("The blob was downloaded to " + sourceFile.getAbsolutePath()))
```
Comment (alzimmermsft) [inline storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java]: `doOnSubscribe` occurs before the file is downloaded, it occurs once the download operation begins, let's reword this to:

```suggestion
                .doOnSubscribe(onSubscribe -> System.out.println("The blob is being downloaded to " + sourceFile.getAbsolutePath()))
```

File: storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java [hunk 3]
```diff
@@ -49,109 +38,72 @@ static File createTempFile() throws IOException {
...
-    static void getBlob(BlockBlobURL blobURL, File sourceFile) {
-        // Get the blob using the low-level download method in BlockBlobURL type
-        // com.microsoft.rest.v2.util.FlowableUtil is a static class that contains helpers to work with Flowable
-        // BlobRange is defined from 0 to 4MB
-        blobURL.download(new BlobRange().withOffset(0).withCount(4*1024*1024L), null, false, null)
-                .flatMapCompletable(response -> {
-                    if(response.rawResponse().request().url().getHost().contains("-secondary"))
-                    {
+    static void getBlob(BlockBlobAsyncClient blockBlobAsyncClient, File sourceFile) {
+        blockBlobAsyncClient.downloadStreamWithResponse(
```
Comment (alzimmermsft) [inline storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java]: As with above, there are APIs for downloading to file as well: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java#L1158

`BlobAsyncClientBase` contains methods that are available in all Blob clients

File: storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java [hunk 4]
```diff
@@ -49,109 +38,72 @@ static File createTempFile() throws IOException {
...
+            if (count > 0) {
+                blobContainerAsyncClient.listBlobs(options).subscribe(response -> {
+                    String output = "Blob name: " + response.getName();
+                    if (response.getSnapshot() != null) {
+                        output += ", Snapshot: " + response.getSnapshot();
+                    }
+                    System.out.println(output);
+                });
+            } else {
+                System.out.println("There are no more blobs to list off.");
```
Comment (alzimmermsft) [inline storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java]: Instead of listing twice here this would be better

```suggestion
        blobContainerAsyncClient.listBlobs(options).subscribe(response -> {
                String output = "Blob name: " + response.getName();
                if (response.getSnapshot() != null) {
                    output += ", Snapshot: " + response.getSnapshot();
                }
                System.out.println(output);
            },
            // Error consumer for if there is an error during blob listing, null is a no-op.
            null,
            // Completion Runnable for when listing completes
            () -> System.out.println("There are no more blobs to list."));
```

Comment (v-hongli1) [reply storage-blobs-java-v10-tutorial-master/src/main/java/tutorial/Tutorial.java]: Fixed in the new version

**Comment by v-xuto** [comment]
@alzimmermsft  We have updated code by your comments, please re-review this PR.

**Review by alzimmermsft — APPROVED** [review]
